.github/workflows/robot/internal/bot/check.go

```diff
@@ -128,9 +133,20 @@ func (b *Bot) reviewersToDismiss(ctx context.Context) ([]string, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	internalApprovals := 0
 	reviewedBy := make(map[string]struct{})
 	for _, review := range reviews {
 		reviewedBy[review.Author] = struct{}{}
+		if review.State == "APPROVED" && b.c.Review.IsInternal(review.Author) {
```

Comment:
if the same author approves twice does it get double counted?

Reply:
Reviews are sorted by time in github.go but they're not deduplicated by author, that's a good point.

We should reuse `review.reviewsByAuthor` here.

Reply:
Ughhh, great catch.

This seems unlikely, but worth fixing.

Reply:
It would only be an issue if an admin approver approved twice

Reply:
Should be fixed either way now.

Reply:
and what happens now if an approver comments after approving (such as https://github.com/gravitational/teleport/pull/14434#pullrequestreview-1037897066)? We have a helper `reviewsByAuthor` which handles that case, could be useful?

Reply:
worst case we won't dismiss reviewers when we could have in this case, approving b/c that's better than what we have now

Reply:
Decided to leave this as is for now to fix the issue at hand.

Worst case, if someone comments after approval, we won't count it, and we may not dismiss reviewers when we could.

This is relatively uncommon, and if it does happen, we'll be no worse off than we are now. Having too many reviewers is better than having not enough.